tsfresh/feature_extraction/data.py

```diff
@@ -291,6 +295,124 @@ def __len__(self):
+    """
+
+    # it = s0, s1, ...,
+    it = iter(seq)
+
+    # islice just takes the first n terms
+    # Therefore it iterates through the first n terms of it
+    # result = (it0, it1, ... itn)
+    result = tuple(itertools.islice(it, window_width))
+
+    # yield the first result
+    if len(result) == window_width:
+        yield result
+        # it = s[n+1], s[n+2], ...
+        # result = (it0, it1, ... itn) = (s0, s1, ..., sn)
+
+    # Iteratively drop the first term of result and append the first elem of 'it' to results
+    for elem in it:
+        result = result[1:] + (elem,)
+        yield result
+
+
+class RollingWideTsFrameAdapter(SliceableTsData):
+
+    def __init__(self, df, column_id, window_width, column_sort=None, value_columns=None):
+        """
+        Rolling window adapter for Pandas DataFrames in wide format, where multiple columns
+        contain different time series for the same id.
+
+        :param df: the data frame
+        :type df: pd.DataFrame
+
+        :param column_id: the name of the column containing time series group ids
+        :type column_id: str
+
+        :param window_width: the width of a single window in the rolling window
+        :type window_width: int
+
+        :param column_sort: the name of the column to sort on
+        :type column_sort: str|None
+
+        :param value_columns: list of column names to treat as time series values.
+            If `None`, all columns except `column_id` and `column_sort` will be used.
+        :type value_columns: list[str]|None
+        """
+
+        # pdb.set_trace()
+        _check_nan(df, column_id)
+
+        if value_columns is None:
+            value_columns = [col for col in df.columns if col not in [column_id, column_sort]]
+
+        if len(value_columns) == 0:
+            raise ValueError("You must provide at least one value column")
+
+        _check_nan(df, *value_columns)
+        _check_colname(*value_columns)
+
+        self.value_columns = value_columns
+
+        self.column_id_dtype = df[column_id].dtype
+
+        if column_sort is not None:
+            _check_nan(df, column_sort)
+            self.df_grouped = df.sort_values([column_sort]).groupby([column_id])
+            self.column_sort = column_sort
+        else:
+            self.df_grouped = df.groupby([column_id])
+            self.column_sort = column_id
+
+        assert (
+            df.groupby(column_id).count().min().min() >= window_width
+        ), "window_width must be equal to or less than the length of the smallest time series"
+
+        self.window_width = window_width
+        self.num_windows = df.groupby(column_id).count().min(axis=1) - window_width + 1
+
+    def __len__(self):
+        return sum([len(self.value_columns) * n_w for n_w in self.num_windows])
+
+    def slice(self, offset, length=None):
+        if 0 < offset >= len(self):
+            raise ValueError("offset out of range: {}".format(offset))
+
+        off_c = 0
```

Comment:
I think some explanatory comment to this block would be good

Reply:
Agreed